modules/system-disaster-recovery/src/test/java/org/apache/ignite/internal/disaster/system/SystemDisasterRecoveryManagerImplTest.java

```diff
@@ -0,0 +1,490 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(WorkDirectoryExtension.class)
+class SystemDisasterRecoveryManagerImplTest extends BaseIgniteAbstractTest {
+    private static final String CLUSTER_NAME = "cluster";
+
+    private static final String NODE_INITIALIZED_VAULT_KEY = "systemRecovery.nodeInitialized";
+    private static final String CLUSTER_NAME_VAULT_KEY = "systemRecovery.clusterName";
+    private static final String RESET_CLUSTER_MESSAGE_VAULT_KEY = "systemRecovery.resetClusterMessage";
+
+    @WorkDirectory
+    private Path workDir;
+
+    private static final String thisNodeName = "node1";
+
+    @Mock
+    private TopologyService topologyService;
+
+    @Mock
+    private MessagingService messagingService;
+
+    private VaultManager vaultManager;
+
+    @Mock
+    private ServerRestarter restarter;
+
+    private final ClusterStateStorage clusterStateStorage = new TestClusterStateStorage();
+    private final ClusterStateStorageManager clusterStateStorageManager = new ClusterStateStorageManager(clusterStateStorage);
+
+    private SystemDisasterRecoveryManagerImpl manager;
+
+    private final ComponentContext componentContext = new ComponentContext();
+
+    private final ClusterNode thisNode = new ClusterNodeImpl(randomUUID().toString(), thisNodeName, new NetworkAddress("host", 1001));
+
+    private final ClusterNode node2 = new ClusterNodeImpl(randomUUID().toString(), "node2", new NetworkAddress("host", 1002));
+    private final ClusterNode node3 = new ClusterNodeImpl(randomUUID().toString(), "node3", new NetworkAddress("host", 1003));
+    private final ClusterNode node4 = new ClusterNodeImpl(randomUUID().toString(), "node4", new NetworkAddress("host", 1004));
+    private final ClusterNode node5 = new ClusterNodeImpl(randomUUID().toString(), "node5", new NetworkAddress("host", 1005));
+
+    private final CmgMessagesFactory cmgMessagesFactory = new CmgMessagesFactory();
+    private final SystemDisasterRecoveryMessagesFactory messagesFactory = new SystemDisasterRecoveryMessagesFactory();
+
+    private final ClusterState usualClusterState = cmgMessagesFactory.clusterState()
+            .cmgNodes(Set.of(thisNodeName))
+            .metaStorageNodes(Set.of(thisNodeName))
+            .version(IgniteProductVersion.CURRENT_VERSION.toString())
+            .clusterTag(randomClusterTag(cmgMessagesFactory, CLUSTER_NAME))
+            .build();
+
+    @Captor
+    private ArgumentCaptor<ResetClusterMessage> resetClusterMessageCaptor;
+
+    @Captor
+    private ArgumentCaptor<NetworkMessage> messageCaptor;
```

Comment:
Should these be fields? I don't think so, they're not used most of the time

Reply:
This allows to avoid the unchecked warnings. And having these fields doesn't seem to be too much of a burden :)

Reply:
They look out of place. What unchecked warnings do you mean? I thought that you instantiate argument captors in a type-safe manner, do you?

Reply:
My comment about type safety was irrelevant (it would be revelant if the captured type was a generic), so I removed these fields.